test/PackageVersionsTest/E2EInstallerTest.php

```diff
@@ -187,6 +187,60 @@ public function testRemovingPluginWithNoDevDoesNotAttemptToGenerateVersions() :
         );
     }
 
+    public function testInstallingPluginWithNoScriptsOverridesOriginalRequirements() : void
+    {
+        $this->createPackageVersionsArtifact();
+        $this->createGenericArtifact(
+            'infection-infection',
+            '0.15.0',
+            [
+                'name' => 'infection/infection',
+                'require' => [
+                    'symfony/process' => '^5.0',
+                ],
+            ]
+        );
+        $this->createGenericArtifact(
+            'symfony-process',
+            '4.0.0',
+            [
+                'name' => 'symfony/process',
+            ]
+        );
+        $this->createGenericArtifact(
+            'symfony-process',
+            '5.0.0',
+            [
+                'name' => 'symfony/process',
+            ]
+        );
+        $this->writeComposerJsonFile(
+            [
+                'name'         => 'package-versions/e2e-transitive',
+                'require'      => [
+                    'ocramius/package-versions' => '1.0.0',
+                    'infection/infection' => '^0.15.0',
+                ],
+                'repositories' => [
+                    ['packagist' => false],
+                    [
+                        'type' => 'artifact',
+                        'url' => $this->tempArtifact,
```

Comment:
If I understand this correctly, the `tempArtifact` contains a full repository manifest here: maybe we should inline the JSON definition to show all existing versions of all packages in here

Reply:
IIUC `tempArtifact` is a foder ([path to folder actually](https://getcomposer.org/doc/05-repositories.md#artifact)) with zip artifacts. composer then chooses the most corrects artifact from present zips